### PR TITLE
Automate compatibility table info

### PR DESCRIPTION
**Pull request by seemywingz**
Add Python Utility to update the compatibility matricoes for tools listed in `static/compatibilities/manifest.yaml`
currently added scrapers and updated compatibility yaml:
- ingress-nginx
- cert-manager
- external-dns

## Test Plan

tested manually, unit test will be added

## Checklist

- [x] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] I have added a meaningful title and summary to convey the impact of this PR to a user.

### CODE REVIEW

File: static/compatibilities/cert-manager.yaml [hunk 1]
```diff
@@ -2,6 +2,10 @@ icon: https://raw.githubusercontent.com/cert-manager/cert-manager/d53c0b9270f8cd
 git_url: https://github.com/cert-manager/cert-manager
 release_url: https://github.com/cert-manager/cert-manager/releases/tag/v{vsn}
 versions:
+- version: 1.15.0
+  kube: ['1.24', '1.25', '1.26', '1.27', '1.28', '1.29', '1.30']
```
Comment (michaeljguarino) [inline static/compatibilities/cert-manager.yaml:6]: why is this one in ascending order while others are descending?

**Review by michaeljguarino — COMMENTED** [review]
Generally want to sort versions in ascending order because the latest version is most significant info

File: utils/compatibility/scrapers/ingress-nginx.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# scrapers/ingress_nginx.py
+
+import requests
+from bs4 import BeautifulSoup
+from collections import OrderedDict
+from utils import printError, update_compatibility_info
+
+
+def scrape():
```
Comment (michaeljguarino) [inline utils/compatibility/scrapers/ingress-nginx.py:9]: This is pretty classic "arrow-code", putting a whole lot of logic in nested if-statements that becomes hard to maintain.  The simple solve is to invert the logic, eg use stuff like:

```python
if response.status_code != 200:
  print_error("could not fetch url")
  return

...rest of logic below
```

Pretty sure the entire function can be re-organized along those lines and require at most one indentation level.

File: static/compatibilities/cert-manager.yaml (hunk 1 above)
Comment (seemywingz) [reply static/compatibilities/cert-manager.yaml:6]: Hmm, I'm not sure; that's definitely weird. I'll look into it.

File: utils/compatibility/scrapers/ingress-nginx.py (hunk 2 above)
Comment (seemywingz) [reply utils/compatibility/scrapers/ingress-nginx.py:9]: Makes sense, I'll refactor this and the other scrapers

File: static/compatibilities/cert-manager.yaml [hunk 3]
```diff
@@ -2,20 +2,24 @@ icon: https://raw.githubusercontent.com/cert-manager/cert-manager/d53c0b9270f8cd
...
+  requirements: []
+  incompatibilities: []
 - version: 1.14.0
-  kube: ['1.29', '1.28', '1.27', '1.26', '1.25', '1.24', '1.24']
+  kube: ['1.30', '1.29', '1.28', '1.27', '1.26', '1.25', '1.24']
   requirements: []
   incompatibilities: []
 - version: 1.13.0
-  kube: ['1.28', '1.27', '1.26', '1.25', '1.24', '1.23']
+  kube: ['1.27', '1.26', '1.25', '1.24', '1.23', '1.22', '1.21']
```
Comment (michaeljguarino) [inline static/compatibilities/cert-manager.yaml:14]: This has to be wrong if 1.12 is compatible up to 1.30 right?

File: utils/compatibility/main.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+
+manifestFile = "../../static/compatibilities/manifest.yaml"
+
+if not os.path.exists(manifestFile):
+    printError(f"Manifest file not found at {manifestFile}")
+else:
+    manifest = readYaml(manifestFile)
+    print()
+    if manifest:
```
Comment (michaeljguarino) [inline utils/compatibility/main.py:27]: this is still arrow-code

File: utils/compatibility/utils.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
+import yaml
+import requests
+
+from collections import OrderedDict
+from colorama import Fore, Style
+from packaging.version import Version
+
+
+def printError(message):
```
Comment (michaeljguarino) [inline utils/compatibility/utils.py:9]: we should standardize on snake casing functions since that is the python standard

**Review by michaeljguarino — APPROVED** [review]

**Pull request merged**